src/parser/Literal.cpp

```diff
@@ -136,12 +136,17 @@ Literal Literal::fromStringRepresentation(std::string internal) {
 }
 
 // __________________________________________
-void Literal::setSubstr(std::size_t start, std::size_t length){
-  auto contentWithoutParentesis = content_.substr(1, beginOfSuffix_ - 2); 
-  auto shortenedContent = ad_utility::getUTF8Substring(contentWithoutParentesis, start, length);
-  auto suffix = content_.substr(beginOfSuffix_);
-  content_ = absl::StrCat("\"", shortenedContent, "\"", suffix);
-  beginOfSuffix_ = content_.size() - suffix.size();
+void Literal::setSubstr(std::size_t start, std::size_t length) {
+  std::size_t contentLength = beginOfSuffix_ - 2;
```

Comment:
For safety we should have `AD_CONTRACT_CHECK`s here that the `start` and  `length` are in the correct ranges
(including letting them fail in the unit tests).